src/bindings/erl_nif/elixir/kdb.ex

```diff
@@ -0,0 +1,277 @@
+
+  def key_unescaped_name(_key) do
+    raise "key_unescaped_name/1 not implemented"
+  end
+
+  def key_get_unescaped_name_size(_key) do
+    raise "key_get_unescaped_name_size/1 not implemented"
+  end
+
+  def key_base_name(_key) do
+    raise "key_base_name/1 not implemented"
+  end
+
+  def key_get_base_name_size(_key) do
+    raise "key_get_base_name_size/1 not implemented"
+  end
+
+  def key_set_base_name(_key, _base_name) do
+    raise "key_set_base_name/2 not implemented"
+  end
+
+  def key_add_base_name(_key, _base_name) do
+    raise "key_add_base_name/2 not implemented"
+  end
+
+  def key_get_namespace(_key) do
+    raise "key_get_namespace/1 not implemented"
+  end
+
+  def key_set_namespace(_key, _ns) do
+    raise "key_set_namespace/2 not implemented"
+  end
+
+  def key_value(_key) do
+    raise "key_value/1 not implemented"
+  end
+
+  def key_get_value_size(_key) do
+    raise "key_get_value_size/1 not implemented"
+  end
+
+  def key_string(_key) do
+    raise "key_string/1 not implemented"
+  end
+
+  def key_set_string(_key, _new_string) do
+    raise "key_set_string/2 not implemented"
+  end
+
+  def key_get_binary(_key) do
+    raise "key_get_binary/1 not implemented"
+  end
+
+  def key_set_binary(_key, _new_binary) do
+    raise "key_set_binary/2 not implemented"
+  end
+
+  def key_lock(_key, _flags) do
+    raise "key_lock/2 not implemented"
+  end
+
+  def key_is_locked(_key) do
+    raise "key_is_locked/1 not implemented"
+  end
+
+  def key_dup(_key, _flags) do
+    raise "key_dup/2 not implemented"
+  end
+
+  # KeySet
+
+  def ks_new(_size) do
+    raise "ks_new/1 not implemented"
+  end
+
+  def ks_dup(_ks) do
+    raise "ks_dup/1 not implemented"
+  end
+
+  def ks_copy(_dest, _source) do
+    raise "ks_copy/2 not implemented"
+  end
+
+  def ks_inc_ref(_ks) do
+    raise "ks_inc_ref/1 not implemented"
+  end
+
+  def ks_dec_ref(_ks) do
+    raise "ks_dec_ref/1 not implemented"
+  end
+
+  def ks_get_ref(_ks) do
+    raise "ks_get_ref/1 not implemented"
+  end
+
+  def ks_clear(_ks) do
+    raise "ks_clear/1 not implemented"
+  end
+
+  def ks_del(_ks) do
+    raise "ks_del/1 not implemented"
+  end
+
+  def ks_get_size(_ks) do
+    raise "ks_get_size/1 not implemented"
+  end
+
+  def ks_append_key(_ks, _key_to_append) do
+    raise "ks_append_key/2 not implemented"
+  end
+
+  def ks_append(_ks, _ks_to_append) do
+    raise "ks_append/2 not implemented"
+  end
+
+  def ks_cut(_ks, _key_cutpoint) do
+    raise "ks_cut/2 not implemented"
+  end
+
+  def ks_pop(_ks) do
+    raise "ks_pop/1 not implemented"
+  end
+
+  def ks_rewind(_ks) do
+    raise "ks_rewind/1 not implemented"
+  end
+
+  def ks_next(_ks) do
+    raise "ks_next/1 not implemented"
+  end
+
+  def ks_current(_ks) do
+    raise "ks_current/1 not implemented"
+  end
+
+  def ks_get_cursor(_ks) do
+    raise "ks_get_cursor/1 not implemented"
+  end
+
+  def ks_set_cursor(_ks, _cursor) do
+    raise "ks_set_cursor/2 not implemented"
+  end
```

Comment:
Old style iterators not needed to implement, see #3976.

```suggestion
```

Reply:
I see. Is the recommended way using an `elektraCursor` and iterating from `0` to `ksGetSize(ks) - 1` using `ksAtCursor`?